client/src/modules/shopping-list/index.jsx

```diff
@@ -23,8 +44,33 @@ class ShoppingList extends Component {
     fetchItemsFromGivenList(id);
   };
 
+  showDialogBox = () => {
+    this.setState({ showDialogBox: true });
+  };
+
+  hideDialogBox = () => {
+    this.setState({ showDialogBox: false });
+  };
+
+  deleteListHandler = (id, redirectCallback) => () => {
+    this.hideDialogBox();
+    const { deleteList } = this.props;
+    deleteList(id, redirectCallback);
```

Comment:
do you need all those 'callback' passing? Could you just do:
deleteList(id).then(_ => this.RedirectHandler())
or just move redirection to deleteList action?